spdx/document.py

```diff
@@ -366,58 +358,52 @@ def validate(self, messages):
         Validate all fields of the document and update the
         messages list with user friendly error messages for display.
         """
-        messages = self.validate_version(messages)
-        messages = self.validate_data_lics(messages)
-        messages = self.validate_name(messages)
-        messages = self.validate_spdx_id(messages)
-        messages = self.validate_namespace(messages)
-        messages = self.validate_ext_document_references(messages)
-        messages = self.validate_creation_info(messages)
-        messages = self.validate_packages(messages)
-        messages = self.validate_extracted_licenses(messages)
-        messages = self.validate_reviews(messages)
-        messages = self.validate_snippet(messages)
-        # messages = self.validate_annotations(messages)
-        # messages = self.validate_relationships(messages)
-
+        if isinstance(messages, list):
```

Comment:
Here you are overriding entirely the messages argument, right? What does this mean then? What's the impact?

Reply:
I think this was in order to keep API compatibility. but this may not make much sense as there might not be that many users we want to avoid braking.

I'll change that to an explicit error